auslib/web/admin/swagger/api.yaml

```diff
@@ -0,0 +1,441 @@
+    allOf:
+      - $ref: "#/definitions/RulesBase"
+    required:
+      - alias
+      - backgroundRate
+      - buildID
+      - buildTarget
+      - channel
+      - comment
+      - data_version
+      - distVersion
+      - distribution
+      - fallbackMapping
+      - headerArchitecture
+      - locale
+      - mapping
+      - osVersion
+      - priority
+      - product
+      - rule_id
+      - systemCapabilities
+      - update_type
+      - version
+      #- whitelist
+
+  RulesBase:
+    title: Rules object definition
+    description: Balrog's Rule Attributes
+    type: object
+
+    required:
+      - update_type # Required during both POSTing and GET(fetch) Operations
+
+    #I was initially hopeful that x-nullable would serve the purpose at class level to define
+    # fields that can expect null values, but my research so
+    # far seems to show that currently x-nullable works at the parameter level but not
+    # within an object/model. Instead declaring 'type' as array like [string, null] or [integer, null]
+
+    # Cannot have additionalProperties as false for strict validation ,else it disallows csrf_token or any other model's properties
+    #additionalProperties: false
+
+    properties:
+      rule_id:
+        description: The id of the rule. This id is necessary to make changes to the rule through the REST API.
+        type: integer
+        format: int32
+        readOnly: true
+        x-nullable: false
+        example: 10
+
+      priority:
+        description: The priority of the rule, relative to other rules. If multiple rules match an incoming request based on the Matchable columns, the rule with the highest priority is chosen.
+        type: ["integer", "null"]
+        format: int32
+        x-nullable: true
+        minimum: 0
+        example: 1
+
+      mapping:
+        description: The Release to construct an update out of if the user is on the right side of a background rate dice roll, or if the background rate is 100. This is a foreign key to the "name" column of the Releases table.
+        type: ["string", "null"]
+        x-nullable: true
+        minLength: 0
+        maxLength: 100
+        example: No-Update
+
+      backgroundRate:
+        description: The percentage of background update requests that if specified. Generally, this is used as a throttle to increase or decrease the rate at which the majority of users receive the latest update.
+        type: ["integer", "null"]
+        format: int32
+        x-nullable: true
+        minimum: 0
+        maximum: 100
+        example: 100
+
+      update_type:
+        description: "The update_type to use in the XML response. It's very rare for a rule to use anything other than 'minor' these days."
+        type: string
+        x-nullable: false
+        minLength: 0
+        maxLength: 15
+        enum: [ "major", "minor"]
+        example: minor
+
+      product:
+        description: The name of the application requesting an update.
```

Comment:
That's the definition of product in the documentation : 
>  Product 	The name of the application requesting an update. 	Exact string match only “Firefox” or “B2G”
How should I describe it ?